File: src/testFile2.ts
import {MemoryInfo, recordMemorySample} from "./types";

export function myFunction2(t: MemoryInfo): Uint8Array | null {
    if (t.foo === 0) {
        return null;
    }

    let len = 0;
    for (let i = 0; i < t.foo; ++i) {
        len += i * i * t.size;
    }
    const mem = new Uint8Array(len);
    if (process.env.RECORD_MEMORY === "true") {
        recordMemorySample(mem);
    }

    return mem;
}

File: src/testFile2.c
#include "thing.h";

void* myFunction2(TestParameter* t) {
    if (t->foo == 0) {
        return 0;
    }

    size_t memory = 0;
    for (int i = 0; i < t->foo; ++i) {
        memory += i * i;
    }

    void* mem = malloc(memory);
    recordMemorySample(mem, memory);

    return mem;
}


